Yafc/Workspace/ProductionTable/ProductionLinkSummaryScreen.cs

```diff
@@ -39,8 +39,92 @@ private void BuildScrollArea(ImGui gui) {
+                GetAllRecipes(page, link.owner)
+                .Select(e => (e, IsParrent(e, parents) ? parentLinks : otherLinks))
+                .Concat(GetAllRecipes(link.owner).Select(e => (e, childLinks)))) {
+                if (link.capturedRecipes.Any(e => e == r)
+                    || (!r.Ingredients.Any(e => e.Goods == link.goods)
+                    && !r.Products.Any(e => e.Goods == link.goods)
+                    && !(r.fuel is not null && r.fuel == link.goods))) {
+                    continue;
+                }
+                float localFlow = DetermineFlow(link.goods, r);
+
+                if ((r.FindLink(link.goods, out var otherLink) && otherLink != link)) {
+                    if (!relationLinks.ContainsKey(otherLink)) {
+                        relationLinks.Add(otherLink, []);
+                    }
+                    table = relationLinks[otherLink];
+                }
+                else {
+                    table = unlinked;
+                }
+
+                if (localFlow > 0) {
+                    table.Add((r, localFlow));
+                }
+                else if (localFlow < 0) {
+                    table.Add((r, localFlow));
+                }
+                else {
+                    table.Add((r, 0));
+                }
+            }
+            var color = 1;
+            if (childLinks.Values.Any(e => e.Any())) {
+                gui.BuildText("Child links: ", Font.subheader);
+                foreach (var relTable in childLinks.Values) {
+                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+                }
+            }
+            if (parentLinks.Values.Any(e => e.Any())) {
+                gui.BuildText("Parent links: ", Font.subheader);
+                foreach (var relTable in parentLinks.Values) {
+                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+                }
+            }
+            if (otherLinks.Values.Any(e => e.Any())) {
+                gui.BuildText("Unrelated links: ", Font.subheader);
+                foreach (var relTable in otherLinks.Values) {
+                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+                }
+            }
+            if (unlinked.Any()) {
+                gui.BuildText("Unlinked: ", Font.subheader);
+                BuildFlow(gui, unlinked, 0, false);
+            }
+        }
+    }
+
+    private bool IsParrent(RecipeRow row, List<ModelObject> parents) => row.Ingredients.Select(e => e.Link).Concat(row.Products.Select(e => e.Link)).Append(row.FuelInformation.Link)
+        .Where(e => e?.ownerObject is not null)
+        .Where(e => e!.goods == link.goods)
+        .Any(e => parents.Contains(e!.ownerObject!));
+
+    private static List<RecipeRow> GetAllRecipes(ProductionTable p, ProductionTable? skip = null) => p == skip ? [] : [..p.recipes, ..p.recipes
```

Comment:
Should this function be part of `ProductionTable`, so you can do `productionTable->GetAllRecipes(skip)` instead?

Reply:
yeah possibly. There was one already that i did not want to mess with, but i probably should. I had to add the `skip` option.

Reply:
I convinced myself i could mess with it and deleted the duplicated code.

Reply:
Can you add the new `skip` option to its function documentation?

Reply:
Yeah, based on the doc it should do a depth-first ordering. Made some more changes to get that property back, idk how important it was but didn't want to mess with it. Also updated the docs.